test/get-module-specifier-test.js

```diff
@@ -59,6 +59,17 @@ describe('get-module-specifier', function() {
     );
   });
 
+  it('identifies named module with a dot delimeter', function() {
+    let moduleExtension = '.hbs';
+
+    assert.equal(getModuleSpecifier(modulePrefix, moduleConfig, 'ui/components/text-editor.glimmer', moduleExtension),
+      'template:/my-app/components/text-editor.glimmer'
```

Comment:
@rwjblue @robbiepitts we should have a broader strategic discussion rather than deciding these related concepts piecemeal across several PRs.